src/Symfony/Bundle/FeatureToggleBundle/Command/FeatureToggleDebugCommand.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FeatureToggleBundle\Command;
+
+use Closure;
+use Symfony\Bundle\FeatureToggleBundle\Debug\TraceableStrategy;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+use Symfony\Component\FeatureToggle\Feature;
+use Symfony\Component\FeatureToggle\Provider\ProviderInterface;
+use Symfony\Component\FeatureToggle\Strategy\OuterStrategiesInterface;
+use Symfony\Component\FeatureToggle\Strategy\OuterStrategyInterface;
+use Symfony\Component\FeatureToggle\Strategy\StrategyInterface;
+use function array_keys;
+use function array_map;
+use function json_encode;
+use function str_repeat;
+use function strlen;
+use function uniqid;
+
+/**
+ * A console command for retrieving information about feature toggles.
+ */
+#[AsCommand(name: 'debug:feature-toggle', description: 'Display configured features and their provider for an application')]
+final class FeatureToggleDebugCommand extends Command
+{
+    /** @var ServiceLocator<ProviderInterface> */
+    private ServiceLocator $featureProviders;
+
+    public function __construct(ServiceLocator $featureProviders)
+    {
+        parent::__construct();
+
+        $this->featureProviders = $featureProviders;
+    }
+
+    protected function configure(): void
+    {
+    }
+
+    /**
+     * @throws \LogicException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $io->title('Feature list grouped by their providers');
+
+        $order = 0;
+        foreach (array_keys($this->featureProviders->getProvidedServices()) as $serviceName) {
+            $featureProvider = $this->featureProviders->get($serviceName);
+
+            ++$order;
+
+            $providerName = $featureProvider::class;
+            if ($providerName !== $serviceName) {
+                $providerName .= " ({$serviceName}).";
+            }
+            $io->section("#{$order} - {$providerName}");
+
+            $tableHeaders = ['Name', 'Description', 'Default', 'Main Strategy'];
+            $tableRows = [];
+
+            foreach ($featureProvider->names() as $featureName) {
+                $feature = $featureProvider->get($featureName);
+
+                $featureGetDefault = Closure::bind(function (): bool {
+                    return $this->default;
+                }, $feature, Feature::class);
```

Comment:
```suggestion
                $featureGetDefault = Closure::bind(fn (): bool => $this->default, $feature, Feature::class);
```

Reply:
ouais je sais... mais je trouvais ca chelou car du coup le $this j'ai vraiment l'impression que c'est la classe et pas bind au truc. Mais je vais update.

Reply:
Ah ouais j'y avais pas pensé. C'est troublant...